spreadsheet/formula/fndatetime.go

```diff
@@ -188,6 +618,40 @@ func TimeValue(args []Result) Result {
 		return MakeErrorResult("TIMEVALUE requires a single string arguments")
 	}
 	timeString := strings.ToLower(args[0].ValueString)
+	if !checkTimeOnly(timeString) { // If date also presents in string, we should validate it first as Excel does
+		_, _, _, timeIsEmpty, errResult := dateValue(timeString)
+		if errResult.Type == ResultTypeError {
+			errResult.ErrorMessage = "Incorrect arguments for TIMEVALUE"
+			return errResult
+		}
+		if timeIsEmpty {
+			return MakeNumberResult(0)
+		}
+	}
+	hours, minutes, seconds, pm, _, errResult := timeValue(timeString)
+	if errResult.Type == ResultTypeError {
+		return errResult
+	}
+	resultValue := daysFromTime(float64(hours), float64(minutes), seconds)
+	if pm {
+		resultValue += 0.5
+	} else if resultValue >= 1 {
+		resultValue -= float64(int(resultValue))
+	}
+	return MakeNumberResult(resultValue)
+}
+
+func checkTimeOnly(timeString string) bool {
+	for _, tf := range timeOnlyFormats {
+		submatch := tf.FindStringSubmatch(timeString)
+		if len(submatch) > 1 {
+			return true
+		}
+	}
+	return false
+}
+
+func timeValue(timeString string) (int, int, float64, bool, bool, Result) {
```

Comment:
Might be useful to comment what the return values are?  Not really clear from looking at this

Reply:
added